runtime/runtime.go

```diff
@@ -441,6 +441,7 @@ func (rt *Runtime) Serve(ctx context.Context) error {
 	rt.serverInitMtx.Lock()
 	rt.serverInitialized = true
 	rt.serverInitMtx.Unlock()
+	rt.Manager.ServerInitialized()
```

Comment:
I think there's a race condition here because there's no guarantee that the socket will have been opened.

The `serverLoop()` call on L430 (above) eventually calls down into `net.Listen` [here](https://github.com/open-policy-agent/opa/blob/main/server/server.go#L429) and at that's what actually calls `socket()` and `listen()` at the OS level (so requests sent in the meantime will get network error.) That said, the window is extremely small; so perhaps that's fine. The solution would be to have this code wait until the server was _actually_ listening for connections but that would require some more invasive changes to the listener interface (e.g., the `serverLoop()` function could return a channel or something to do indicate the listener was opened...)

@gshively11 if you agree w/ the above and are okay with the potential race condition, just include a comment here.

Reply:
Ah, good catch.  I think we're probably OK with this race condition, I'll add the comment.  If it does end up being an issue, a modification as you described shouldn't be a breaking change, so it should be easy to implement if we need it.